crates/rspack_core/src/utils/queue.rs

```diff
@@ -1,14 +1,34 @@
 use std::collections::VecDeque;
 
-#[derive(Default)]
-pub struct WorkerQueue<T> {
+use rustc_hash::FxHashMap;
+use tokio::sync::mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender};
+
+use crate::{Compilation, ModuleIdentifier};
+
+pub struct WorkerQueue<T, K> {
   inner: VecDeque<T>,
+  queue_tx: QueueHandler<T, K>,
```

Comment:
Is it possible for us to remove old `VecDeque` and just use `mpsc` here?

Reply:
If we use mpsc exclusively, I'm afraid the performance will decrease.

Reply:
I have the same concern here. So I'm just wondering is it possible to only support adding handler for a certain queue or we have to implement them all.

Reply:
For now, only used 3 of 5, factorize_queue, process_dependencies_queue and build_time_execution_queue.
The reason I use separate queue_handler before, is that we can check only one channel at every iteration, instead of checking 5 channels in this PR, let me see benchmark result and then try optimizing a little bit

Reply:
The other concern that I have other than performance is that exposing every queue to the outside may break the contract which is provided by `handle_module_creation`. As this method is the entry point of building a module. We'd better use a semantic function instead of modifying the queue directly (If you can abstract the behavior as much as possible).

Reply:
Understand, I think it's low level part of Webpack, users should be aware of what they are doing if they want to use this